src/main/java/fr/rader/boblite/Main.java

```diff
@@ -23,242 +29,123 @@ public void start() {
-                    // reason == 2 -> begin raining
-                    // reason == 7 -> rain level change
-                    // reason == 8 -> thunder level change
-                    // iirc reason 1 and 2 are swapped
-                    if (reason != 1 && reason != 2 && reason != 7 && reason != 8) {
-                        // if the packet does not change the rain, we write it
-                        // packet header
-                        writer.writeInt(timestamp);
-                        writer.writeInt(size);
-                        writer.writeVarInt(packetID);
-
-                        // packet content
-                        writer.writeByte(reason);
-                        writer.writeFloat(reader.readFloat());
-                    } else {
-                        // we skip the next 4 bytes (new rain/thunder level)
-                        reader.skip(4);
-                    }
+        if (tempFileDirectory == null) {
+            System.exit(0);
+        }
 
-                    continue;
-                }
+        // Okay, time to get working!
+
+        // Get the total number of system CPU threads.
+        int totalSystemThreadNumber = Runtime.getRuntime().availableProcessors();
+        int executorThreadNumber;
+        if (replays.size() < totalSystemThreadNumber){
+            executorThreadNumber = replays.size();
+        }else{
+            // Leave at least one system thread idle, don't make the computer unusable.
+            executorThreadNumber = totalSystemThreadNumber - 1;
+            // If for whatever reason the computer only has one CPU core/thread, just use one thread.
+            if(executorThreadNumber < 1) executorThreadNumber = 1;
+        }
 
-                // we write the packet timestamp, size, packet id
-                writer.writeInt(timestamp);
-                writer.writeInt(size);
-                writer.writeVarInt(packetID);
-                // and finally, we write the packet data
-                writer.writeByteArray(reader.readFollowingBytes(size - 1));
-            }
+        System.out.println("Using " + executorThreadNumber + " thread(s) for " + replays.size() + " task(s)...");
 
-            // we then flush the writer
-            writer.flush();
+        ExecutorService executor = Executors.newFixedThreadPool(totalSystemThreadNumber);
 
-            // changing the progress bar action text,
-            // as we're now doing something else
-            progressBar.setActionText("Writing the replay, this can take a while...");
+        long startTime = System.currentTimeMillis();
 
-            // we open the zip file, write the recording.tmcpr file and close the zip
-            replayZip.open();
-            replayZip.addFile(writer.getInputStream(), "recording.tmcpr");
-            replayZip.close();
+        // Submit all tasks.
+        for(ReplayData replay : replays){
+            executor.submit(new EditReplayTask(
```

Comment:
Move the `new EditReplayTask(` to a new line

Reply:
Resolved in commit [fix code formatting - 63d35e3](https://github.com/RaderRMT/BobLite/pull/1/commits/63d35e385e87abea775c880b09dfc2494ea08a84)